tests/data_structures/union_find/test_union_find.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import absolute_import
+from data_structures.union_find.union_find import UnionFind
+import pytest
+
+def test_init_with_valid_size():
+    '''Testing  for set with valid size'''
+    uf = UnionFind(5)
+    assert uf.size == 5
+
+def test_init_with_invalid_size_1():
+    '''Testing  for set with invalid size'''
+    uf = UnionFind(5)
+    with pytest.raises(ValueError):
+        uf = UnionFind(0)
+            
+def test_init_with_invalid_size_2():
+    '''Testing  for set with invalid size'''
+    uf = UnionFind(5)
+    with pytest.raises(ValueError):
+        uf = UnionFind(-5)
+
+def test_union_with_valid_values():
+    '''Testing union for valid set of values'''
+    uf = UnionFind(10)
+
+    for i in range(11):
+        for j in range(11):
+            uf.union(i, j)
+
+def test_union_with_invalid_values_1():
+    '''Testing union for invalid set of values'''
+    uf = UnionFind(10)
+    with pytest.raises(ValueError):
+        uf.union(-1, 1)
+            
+def test_union_with_invalid_values_2():
+    '''Testing union for invalid set of values'''
+    uf = UnionFind(10)
+    with pytest.raises(ValueError):
+        uf.union(11, 1)
+
+def test_same_set_with_valid_values_1():
+    '''Testing union for valid set of values with the same set'''
+    uf = UnionFind(10)
+
+    for i in range(11):
+        for j in range(11):
+            if i == j:
+                assert uf.same_set(i, j) == True
+            else:
+                assert uf.same_set(i, j) == False
+
+def test_same_set_with_valid_values_2():
+    '''Testing union for valid set of values with the same set'''
```

Comment:
Replace all
`assert something == True` with `assert something` and
`assert something == False` with `assert not something` in this test.